backend/dataall/cdkproxy/assets/datalakelocationcustomresource/index.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import os
+import json
+import boto3
+from botocore.exceptions import ClientError
+
+logger = logging.getLogger()
+logger.setLevel(os.environ.get("LOG_LEVEL", "INFO"))
+log = logging.getLogger(__name__)
+
+lf_client = boto3.client("lakeformation", region_name=os.environ.get("AWS_REGION"))
+
+
+def on_event(event, context):
+    request_type = event["RequestType"]
+    if request_type == "Create":
+        return on_create(event)
+    if request_type == "Update":
+        return on_update(event)
+    if request_type == "Delete":
+        return on_delete(event)
+    raise Exception(f"Invalid request type: {request_type}")
+
+
+def on_create(event):
+    """ Checks if the S3 location is already registered in Lake Formation.
+    If already registered it updated the roleArn for the location.
+    If not registered, it registers the location.
+    """
+    props = event["ResourceProperties"]
+    if not _is_resource_registered(props["ResourceArn"]):
+        register(props)
+    else:
+        update(props)
+
+
+def _is_resource_registered(resource_arn: str):
+    lf_resources = lf_client.list_resources(FilterConditionList=[{"Field": "RESOURCE_ARN", "ComparisonOperator": "EQ", "StringValueList": [resource_arn]}])
+    return len(lf_resources["ResourceInfoList"]) > 0
+
```

Comment:
@dosiennik I found out that when we have more than 100 registered locations the pagination of this api call prevents us from seeing the registered resource. It tries to re-register it and it fails :( 
So, let's say we have around 500 registered locations and a particular dataset is returned in the 4th page when we list_resources without any filter condition. We are updating a dataset, the handler is triggered and the list_resources with filter returns nothing as it is trying to filter from the first page of results.

I am working on a fix for this, I think I am going to try to register the resource and check the exception

Reply:
oh, that's interesting, thanks for the finding! we are actually applying a filter 
`lf_client.list_resources(FilterConditionList=[{"Field": "RESOURCE_ARN", "ComparisonOperator": "EQ", "StringValueList": [resource_arn]}])`

but maybe it works another way under the hood

Reply:
Exactly, I think that filter is only filtering the results from the first page of results. I have tested it and it seems to be the case because NextToken is returned.

```
[DEBUG]	2023-04-03T08:05:38.808Z	1c09bb21-65be-47fd-9de2-29a77d273c02	Response body:
b'{
    "NextToken": "eyJsYXN0RXZhbHVhdGVkS2V5Ijp7IkhBU0hfS0VZIjp7InMiOiI4NTUyMjU2OTU4NzlfODU1MjI1Njk1ODc5In0sIlJBTkdFX0tFWSI6eyJzIjoiczM6Ly9kYXRhYWxsLWFwaWRhdGFzZXR5bGltaXQtMTkyLTlyYnE3bGpvIn19LCJleHBpcmF0aW9uIjp7InNlY29uZHMiOjE2ODA1OTU1MzgsIm5hbm9zIjo4MDUwMDAwMDB9fQ==",
    "ResourceInfoList": []
}'

```

Reply:
whoops, so I need to take a look as well, for now I have no other idea than iterating page by page until the next token is returned from the response :/

Reply:
That is what I just pushed :)
Take a look at the logs, this is for an environment with around 500 datasets. Until the fourth call it does not find the resource:
![image](https://user-images.githubusercontent.com/71252798/229497518-d6ed5064-3ca8-4285-aabd-cf4edcf36f27.png)